src/Datadog.Trace.ClrProfiler.Managed/Integrations/HttpMessageHandlerIntegration.cs

```diff
@@ -19,11 +18,18 @@ public static class HttpMessageHandlerIntegration
         private const string SystemNetHttp = "System.Net.Http";
         private const string Major4 = "4";
 
-        private const string HttpMessageHandler = "System.Net.Http.HttpMessageHandler";
-        private const string HttpClientHandler = "System.Net.Http.HttpClientHandler";
+        private const string HttpMessageHandlerTypeName = "HttpMessageHandler";
+        private const string HttpClientHandlerTypeName = "HttpClientHandler";
+
+        private const string HttpMessageHandler = SystemNetHttp + "." + HttpMessageHandlerTypeName;
+        private const string HttpClientHandler = SystemNetHttp + "." + HttpClientHandlerTypeName;
         private const string SendAsync = "SendAsync";
 
         private static readonly Vendors.Serilog.ILogger Log = DatadogLogging.GetLogger(typeof(HttpMessageHandlerIntegration));
+        private static readonly string[] NamespaceAndNameFilters = { ClrNames.GenericTask, ClrNames.HttpRequestMessage, ClrNames.CancellationToken };
+
+        private static Type _httpMessageHandlerResultType;
+        private static Type _httpClientHandlerResultType;
```

Comment:
There may be an edge case here where we end up with multiple versions of `HttpMessageHandler` or `HttpClientHandler` loaded into the app domain. Just to make sure I understand, is this what you're checking for below in `taskResultType.Assembly != httpMessageHandler.Assembly`?

Reply:
I preferred to play safe yes. Worst case scenario this will revert the optimization and we'll be back to fetching the type every time